modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -185,4 +190,31 @@ public void onFailure(Exception t) {
         }
         new RetryHelper().run();
     }
+
+    /**
+     * Wrap the ResponseException in an exception that'll preserve its status code if possible so we can send it back to the user. We might
+     * not have a constant for the status code so in that case we just use 500 instead. We also extract make sure to include the response
+     * body in the message so the used can figure out *why* the remove Elasticsearch service threw the error back to us.
+     */
+    static ElasticsearchStatusException wrapExceptionToPreserveStatus(int statusCode, @Nullable HttpEntity entity, Exception cause) {
+        RestStatus status = RestStatus.fromCode(statusCode);
+        String messagePrefix = "";
+        if (status == null) {
+            messagePrefix = "Couldn't extract status [" + statusCode + "]. ";
+            status = RestStatus.INTERNAL_SERVER_ERROR;
+        }
+        try {
+            String message;
+            if (entity == null) {
+                message = messagePrefix + "No error body.";
```

Comment:
I did that at first but it felt so weird for an error not to have a body that I found myself expecting *some* message when writing the tests.